Tests/RingBufferTestsKiwi.m

```diff
@@ -0,0 +1,129 @@
+                _RPTMeasurement *measurement = [buffer measurementAtIndex:validIndex];
+                [[theValue(measurement.trackingIdentifier) should] equal:theValue(validIndex)];
+            });
+
+            it(@"should return a nil measurement when the passed index is bigger than the buffer's size", ^{
+                _RPTMeasurement *measurement = [buffer measurementAtIndex:size + 1];
+                [[measurement should] beNil];
+            });
+
+            it(@"should return a nil measurement when the passed index is equal to the buffer's size", ^{
+                _RPTMeasurement *measurement = [buffer measurementAtIndex:size];
+                [[measurement should] beNil];
+            });
+        });
+
+        context(@"measurementWithTrackingIdentifier", ^{
+
+            it(@"should return a non-nil measurement when the passed an index is less than the buffer's size", ^{
+                NSUInteger validIndex = size-1;
+                _RPTMeasurement *measurement = [buffer measurementWithTrackingIdentifier:validIndex];
+                [[measurement should] beNonNil];
+            });
+
+            it(@"should return a measurement having the same tracking identifier as expected", ^{
+                NSUInteger validIndex = size-1;
+                _RPTMeasurement *measurement = [buffer measurementWithTrackingIdentifier:validIndex];
+                [[theValue(measurement.trackingIdentifier) should] equal:theValue(validIndex)];
+            });
+
+            it(@"should return a nil measurement when the passed index is bigger than the buffer's size", ^{
+                _RPTMeasurement *measurement = [buffer measurementWithTrackingIdentifier:size + 1];
+                [[measurement should] beNil];
+            });
+
+            it(@"should return a nil measurement when the passed index is equal to the buffer's size", ^{
+                _RPTMeasurement *measurement = [buffer measurementWithTrackingIdentifier:size];
+                [[measurement should] beNil];
+            });
+        });
+    });
+
+    describe(@"nextMeasurement", ^{
+
+        it(@"should return a non-nil measurement when the buffer is not full", ^{
+            _RPTRingBuffer *buffer = [[_RPTRingBuffer alloc] initWithSize:10];
+            _RPTMeasurement *measurement = nil;
+            for (int i = 0; i < 5; i++) {
+                measurement = [buffer nextMeasurement];
+            }
+            [[measurement should] beNonNil];
+            [[theValue(measurement.trackingIdentifier) should] equal:theValue(5)];
+        });
+
+        it(@"should return a nil measurement when the buffer is full", ^{
+            _RPTRingBuffer *buffer = [[_RPTRingBuffer alloc] initWithSize:10];
+            _RPTMeasurement *measurement = nil;
+            for (int i = 0; i < 11; i++) {
+                measurement = [buffer nextMeasurement];
+            }
+            [[measurement should] beNil];
+        });
```

Comment:
We could also test the scenario where the buffer has been filled, then partially processed e.g. index 0-5 are cleared but 6-9 are filled, calling get nextMeasurement should succeed.